src/pool/rpc.h

```diff
@@ -510,16 +471,478 @@ CRT_RPC_DECLARE(pool_tgt_discard, DAOS_ISEQ_POOL_TGT_DISCARD,
 
 static inline int
 pool_req_create(crt_context_t crt_ctx, crt_endpoint_t *tgt_ep, crt_opcode_t opc,
-		crt_rpc_t **req)
+		const uuid_t pi_uuid, const uuid_t pi_hdl, uint64_t *req_timep, crt_rpc_t **req)
 {
-	crt_opcode_t opcode;
+	int                    rc;
+	crt_opcode_t           opcode;
+	static __thread uuid_t cli_id;
+	struct pool_op_in     *in;
+
+	if (uuid_is_null(cli_id))
+		uuid_generate(cli_id);
 
 	opcode = DAOS_RPC_OPCODE(opc, DAOS_POOL_MODULE,
 				 dc_pool_proto_version ? dc_pool_proto_version : DAOS_POOL_VERSION);
 	/* call daos_rpc_tag to get the target tag/context idx */
 	tgt_ep->ep_tag = daos_rpc_tag(DAOS_REQ_POOL, tgt_ep->ep_tag);
 
-	return crt_req_create(crt_ctx, tgt_ep, opcode, req);
+	rc = crt_req_create(crt_ctx, tgt_ep, opcode, req);
+	if (rc != 0)
+		return rc;
+	in = crt_req_get(*req);
+	uuid_copy(in->pi_uuid, pi_uuid);
+	if (uuid_is_null(pi_hdl))
+		uuid_clear(in->pi_hdl);
+	else
+		uuid_copy(in->pi_hdl, pi_hdl);
+
+	if (req_timep && (*req_timep == 0))
+		*req_timep = d_hlc_get();
+
+	/* Temporary req_timep check: some opcodes aren't (yet) at v8 and don't have the op key */
+	if (req_timep && (dc_pool_proto_version >= POOL_PROTO_VER_WITH_SVC_OP_KEY)) {
+		struct pool_op_v6_in *in6 = crt_req_get(*req);
+
+		uuid_copy(in6->pi_cli_id, cli_id);
+		in6->pi_time = *req_timep;
+	}
+	return 0;
+}
+
+static inline void
```

Comment:
in updating this PR I forgot to move these functions into the neighborhood of the respective definitions above. If there is a need to update this PR for other review comments I can do that in the next push. Otherwise I can get it in a subsequent PR for this feature development.